components/suggest/src/suggestion.rs

```diff
@@ -33,6 +33,20 @@ pub enum Suggestion {
         raw_click_url: String,
         score: f64,
     },
+    AmpMobile {
```

Comment:
Since AMP suggestions share the same structure on desktop and mobile, I think we can use `Amp` for both. Let's add a doc string to to the `Amp` variant to note this.

Reply:
Hmmm, I'm not sure about this one! Right now, there's a 1:1 correspondence between the variants of `SuggestionProvider` and `Suggestion`—so if you ask for `SuggestionProvider::Amp` and `SuggestionProvider::Wikipedia`, you'll get `Suggestion::Amp` and `Suggestion::Wikipedia` back.

Even though the AMP mobile suggestions have the same structure, I think passing `SuggestionProvider::AmpMobile` and `SuggestionProvider::Wikipedia`, and getting `Suggestion::Amp` and `Suggestion::Wikipedia` back, feels a bit too special case-y for me 🤔

Reply:
I thought a bit more about this one, and now I like @ncloudioj's suggestion more :joy:

There's no reason why a `SuggestionProvider` couldn't return different types of `Suggestion`s, and I could see us doing this in the future.